wsbroadcastserver/wsbroadcastserver.go

```diff
@@ -117,8 +137,52 @@ func (s *WSBroadcastServer) Start(ctx context.Context) error {
 
 		safeConn := deadliner{conn, s.settings.IOTimeout}
 
+		// Prepare handshake header writer from http.Header mapping.
+		header := ws.HandshakeHeaderHTTP(http.Header{
+			HTTPHeaderFeedServerVersion: []string{strconv.Itoa(FeedServerVersion)},
+			HTTPHeaderChainId:           []string{strconv.FormatUint(s.chainId, 10)},
+		})
+
+		var feedClientVersionSeen bool
+		var requestedSeqNum arbutil.MessageIndex
+		upgrader := ws.Upgrader{
+			OnHeader: func(key []byte, value []byte) error {
+				headerName := string(key)
+				if headerName == HTTPHeaderFeedClientVersion {
+					feedClientVersion, err := strconv.ParseUint(string(value), 0, 64)
+					if err != nil {
+						return err
+					}
+					if feedClientVersion < FeedClientVersion {
+						return ws.RejectConnectionError(
+							ws.RejectionStatus(http.StatusBadRequest),
+							ws.RejectionReason(fmt.Sprintf("Feed Client version too old: %d, expected %d", feedClientVersion, FeedClientVersion)),
+						)
+					}
+					feedClientVersionSeen = true
+				} else if headerName == HTTPHeaderRequestedSequenceNumber {
+					num, err := strconv.ParseUint(string(value), 0, 64)
+					if err != nil {
+						return fmt.Errorf("unable to parse HTTP header key: %s, value: %s", headerName, string(value))
+					}
+					requestedSeqNum = arbutil.MessageIndex(num)
+				}
+
+				return nil
+			},
+			OnBeforeUpgrade: func() (ws.HandshakeHeader, error) {
+				if s.settings.RequireVersion && !feedClientVersionSeen {
+					return nil, ws.RejectConnectionError(
+						ws.RejectionStatus(http.StatusBadRequest),
+						ws.RejectionReason("Feed-Client-Version HTTP header missing"),
```

Comment:
If we change the header name, we should change it here too (or maybe just pull it in from the variable here)

Reply:
:+1: